golem-worker-service-base/src/gateway_execution/gateway_binding_resolver.rs

```diff
@@ -36,30 +42,37 @@ pub trait GatewayBindingResolver<Namespace, ApiDefinition> {
     async fn resolve_gateway_binding(
         &self,
         api_definitions: Vec<ApiDefinition>,
-    ) -> Result<ResolvedGatewayBinding<Namespace>, GatewayBindingResolverError>;
+    ) -> Result<ResolvedGatewayBinding<Namespace>, ErrorOrRedirect>;
 }
 
 #[derive(Debug)]
-pub enum GatewayBindingResolverError {
-    RibInputTypeMismatch(RibInputTypeMismatch),
-    Internal(String),
-    RouteNotFound,
+pub enum ErrorOrRedirect {
+    Error(GatewayBindingResolverError),
+    Redirect(poem::Response),
 }
 
-impl GatewayBindingResolverError {
-    pub fn internal(err: &str) -> Self {
-        GatewayBindingResolverError::Internal(err.to_string())
+impl ErrorOrRedirect {
```

Comment:
During the binding resolution phase itself (resolving a binding based on a http request, where the worker name is also resolved (Ex: worker-${request.auth.given_name})), middlewares such as authentication is applied, and can decide to redirect or not. That's the main change.